app/models/changeset.rb

```diff
@@ -28,6 +28,20 @@ def commits
     @commits ||= comparison.commits.map {|data| Commit.new(repo, data) }
   end
 
+  def last_commit_status
+    if comparison.commits.nil? || comparison.commits.empty?
+      StatusResult.null_result
+    else
+      ref = comparison.commits.last["sha"]
+      state = Rails.cache.fetch(status_cache_key) do
```

Comment:
let's give this an expiration time for the case when a flaky build is re-run and passes